src/moscot/plotting/_utils.py

```diff
@@ -359,45 +360,137 @@ def _plot_temporal(
+            else:
+                vmin = 0
+                vmax = 1
+            adata.obs[f"result_key_{i}"] = (tmp - vmin) / (vmax - vmin)
+            size = (mask * 120000 * dot_scale_factor + (1 - mask) * 120000) / adata.n_obs  # 120,000 from scanpy
+
+            _ = kwargs.pop("color_map", None)
+            _ = kwargs.pop("palette", None)
+
+            if isinstance(cont_cmap, str):
+                try:
+                    cont_cmap = mpl.colormaps[cont_cmap]
+                except KeyError:
+                    raise KeyError("`cont_cmap` is not a valid key for `matplotlib.colormaps`.")
+            if (time_points[i] == start and push) or (time_points[i] == end and not push):
+                vmin, vmax = np.nanmin(adata.obs[f"result_key_{i}"]), np.nanmax(adata.obs[f"result_key_{i}"])
+                adata.obs[f"result_key_{i}"] = adata.obs[f"result_key_{i}"].astype("str")
+                st = f"not in {time_points[i]}"
+                adata.obs[f"result_key_{i}"] = adata.obs[f"result_key_{i}"].replace("nan", st)
+                palette = {str(vmax): cont_cmap.reversed()(0), str(vmin): cont_cmap(0), st: na_color}
+                kwargs["palette"] = palette
+            else:
+                kwargs["color_map"] = cont_cmap
+
+        sc.pl.embedding(
+            adata=adata,
+            basis=basis,
+            color=f"result_key_{i}",
+            title=titles[i],
+            size=size,
+            ax=ax,
+            show=show,
+            na_color=na_color,
+            **kwargs,
+        )
+    if suptitle is not None:
+        fig.suptitle(suptitle, fontsize=suptitle_fontsize)
     if save:
         fig.figure.savefig(save, bbox_inches="tight")
     return fig
+
+
+def _color_transition(c1: str, c2: str, num: int, alpha: float) -> List[str]:
+    if not mpl.colors.is_color_like(c1):
+        raise ValueError(f"{c1} cannot be interpreted as an RGB color.")
+    if not mpl.colors.is_color_like(c2):
+        raise ValueError(f"{c2} cannot be interpreted as an RGB color.")
+    c1_rgb = np.array(mpl.colors.to_rgb(c1))
+    c2_rgb = np.array(mpl.colors.to_rgb(c2))
+    return [mpl.colors.to_rgb((1 - n / num) * c1_rgb + n / num * c2_rgb) + (alpha,) for n in range(num)]
+
+
+def _create_col_colors(adata: AnnData, obs_col: str, subset: Union[str, Sequence[str]]) -> Optional[mcolors.Colormap]:
+
+    if isinstance(subset, Sequence) and len(subset) == 1 and isinstance(subset[0], str):
+        subset = subset[0]
+    if not is_categorical_dtype(adata.obs[obs_col]):
+        raise TypeError(f"`adata.obs['{obs_col}'] must be of categorical type.")
+    if isinstance(subset, str):
+        try:
+            colorDict = {
+                cat: adata.uns[f"{obs_col}_colors"][i] for i, cat in enumerate(adata.obs[obs_col].cat.categories)
+            }
+        except KeyError:
+            raise KeyError(f"Unable to access `adata.uns['{obs_col}_colors']`.") from None
+
+        color = colorDict[subset]
```

Comment:
Why are you creating the full `colorDict` when in the end you're extracting only 1 value?